Update readme (#46)
terraref/traitvis-webapp: README.md

Time series box, line plots and heatmaps for every available season

> ### Set up a local development environment

Reviewer: This could go under a section `## Development`

Then explain at a high level what steps will be required. (R, R libraries, other dependencies)

#### Database connection

This application requires a connection to a BETYdb database. It requires a BETYdb database that uses the experiments table to associate time ranges and sites or plots with specific experiments or seasons.

A connection to a local instance of BETYdb can be created using the TERRA-REF BRAPI API. Clone the [terraref/brapi repository](https://github.com/terraref/brapi) and run Docker commands to create a connection. 

> Clone terraref/brapi repository:

Reviewer: Would it make sense to just put a copy of the docker-compose.override.yml from brapi into this repository? It seems like a lot of work to clone a whole other repository just to get the docker compose file.

Author: @dlebauer yes, I think it would be a good idea to put the copy of the docker-compose.override.yml in the brapi repository.

But, I am confused about the second part of your comment. Wouldn't the user have to clone the brapi repository to set up the brapi docker in order to connect to the database? Is there another way to do it?

Reviewer: Have you tried to run the commands with the docker-compose file? What if you just removed lines 42:60 of https://github.com/terraref/brapi/blob/master/docker-compose.yml#L42?

Author: The commands in the docker-compose file work (without removing any lines). Instead of cloning the brapi repo, I will copy the docker-compose.yml and docker-compose.override.yml into this repo.

Reviewer: The lines for running brapi and brava don't make sense in this repo and should be removed to avoid confusion.

Author: Okay

```sh
git clone https://github.com/terraref/brapi.git
cd brapi
```

Within the brapi folder, create a `docker-compose.override.yml` file to expose a port on your local machine. The following chunk can be used to expose port 5432 and should be added to the file:

```sh
version: '3'
services:
  postgres:
    ports:
      - '5432:5432'
```

> Note: If port 5432 is being used on your computer, you can use any other number. Just make sure to change the first number in the pairing to the number you choose.

Reviewer: ```suggestion
Note: If you are already running Postgres, it is likely that port 5432 is being used on your computer. 
In this case, you can assign another number such as 5433. 
Just make sure to change the first number in the pairing to the number you choose, e.g. 5433:5432 means you are mapping your computer's port 5433 to the Docker container's Postgres port 5432.
```

Initialize the database:

> 

Reviewer: (note: ignore the permission errors)

```sh

> docker-compose up -d postgres

Reviewer: ```suggestion
# note: ignore the permission errors
docker-compose up -d postgres
```

docker-compose run --rm bety initialize

> docker-compose run --rm bety sync

Reviewer: add note ... ignore permission errors

```

#### R packages

> This application requires the following R packages:

Reviewer: ```suggestion
This application depends on a number of R packages; this is how you can install them:
```

> `shiny`, `shinythemes`, `scales`, `lubridate`, `dplyr`, `ggplot2`, `timevis`, `rgeos`, `leaflet`, `cronR`, `stringr`, `kableExtra`, `raster`, `mapview`, `leafem`, and `sf`.

Reviewer: Probably easier to list them at the same time that you show how to install them:

```r
install.packages(c('shiny', 'shinythemes', 'lubridate', 'dplyr', 'ggplot2', 'timevis', 'rgeos', 'leaflet', 'cronR', 
                                'stringr', 'kableExtra', 'raster', 'mapview', 'leafem', 'sf'))
```

#### Download image thumbnails

Fullfield image thumbnails are available to be displayed under heatmaps. These thumbnails can be downloaded from the `/ua-mac/Level_2/rgb_fullfield/_thumbs/` endpoint on [Globus](https://www.globus.org/).

These thumbs should be saved to the following path in your home directory `~/data/terraref/sites/ua-mac/Level_2/rgb_fullfield/_thumbs`.

#### How to run Shiny Application

Clone the [terraref/traitvis](https://github.com/terraref/traitvis-webapp) repository:

```sh
git clone https://github.com/terraref/traitvis-webapp.git
cd traitvis-webapp
```

The `cache-refresh.R` script needs to be run and requires a database connection. To connect to the local instance of BETYdb that you set up above, you will need to set the following connection parameters as environment variables.

In your R console, run the following commands:

```sh
Sys.setenv(bety_host = 'localhost')

> Sys.setenv(bety_port = '5433')

Reviewer: above when deploying docker this is set to 5432; perhaps it should be set to 5433 above?

Author: Yes it should. I have added a docker-compose.override.yml to the repo that exposes port 5433.
